File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        n = len(candies)
        max = 0 
        for i in range(n):
            if max < candies[i]:
                max = candies[i]

        
        result = []

        for j in range(n):
            if candies[j] + extraCandies >= max:
                result.append(True)
            else:
                result.append(False)
        
        return result

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = 0;
        int n = candies.length;
        for (int i=0; i<n; i++){
            if(max < candies[i]){
                max = candies[i];
            }
        }

        List<Boolean> result = new ArrayList<>(n);

        for ( int i = 0; i < n; i++){
            if(candies[i]+extraCandies >= max){
                result.add(true);
            }
            else{
                result.add(false);
            }
        }

        return result;
    }
}